nautobot/ipam/views.py

```diff
@@ -661,6 +677,98 @@ def post(self, request):
+                template_name="ipam/ipaddress_merge.html",
+                context={
+                    "queryset": queryset,
+                    "return_url": self.get_return_url(request),
+                },
+            )
+        else:
+            msg = "No duplicate IPs found."
+            messages.warning(request, msg)
+            return redirect(self.get_return_url(request))
+
+    def post(self, request):
+        logger = logging.getLogger(__name__)
+        collapsed_ips = IPAddress.objects.filter(pk__in=request.POST.getlist("pk"))
+        with transaction.atomic():
+            try:
+                _, deleted_info = collapsed_ips.delete()
+                deleted_count = deleted_info[IPAddress._meta.label]
+            except ProtectedError as e:
+                logger.info("Caught ProtectedError while attempting to delete objects")
+                handle_protectederror(collapsed_ips, request, e)
+                return redirect(self.get_return_url(request))
+            merged_attributes = request.POST
+
+            if merged_attributes.get("namespace"):
+                namespace = Namespace.objects.get(pk=merged_attributes.get("namespace"))
+            else:
+                namespace = None
+            if merged_attributes.get("tenant"):
+                tenant = Tenant.objects.get(pk=merged_attributes.get("tenant"))
+            else:
+                tenant = None
+            if merged_attributes.get("status"):
+                status = Status.objects.get(pk=merged_attributes.get("status"))
+            else:
+                status = None
+            if merged_attributes.get("role"):
+                role = Role.objects.get(pk=merged_attributes.get("role"))
+            else:
+                role = None
+            if merged_attributes.get("tags"):
+                tag_pk_list = merged_attributes.get("tags").split(",")
+                tags = Tag.objects.filter(pk__in=tag_pk_list)
+            else:
+                tags = None
+            merged_ip = IPAddress.objects.create(
+                address=merged_attributes.getlist("address")[0],
+                mask_length=merged_attributes.get("mask_length"),
+                namespace=namespace,
+                tenant=tenant,
+                status=status,
+                role=role,
+                description=merged_attributes.get("description"),
+            )
+            if tags:
+                merged_ip.tags.set(tags)
+            msg = f"Merged {deleted_count} {self.queryset.model._meta.verbose_name}"
+            if hasattr(merged_ip, "get_absolute_url"):
+                msg = f'{msg} into <a href="{merged_ip.get_absolute_url()}">{escape(merged_ip)}</a>'
+            else:
+                msg = f"{msg} into {escape(merged_ip)}"
+            logger.info(msg)
+            messages.success(request, mark_safe(msg))
+        host_values = self.queryset.values("host").order_by().annotate(count=models.Count("host")).filter(count__gt=1)
```

Comment:
If we add the ability (which I think is needed) to let the user opt-out of merging certain IPs, then this probably needs to be "smarter" and do something like `self.queryset.filter(host__gte=<most recently processed IP>)....` so that it doesn't go into a loop when the user doesn't merge certain records.